Marlin/src/Marlin.cpp

```diff
@@ -320,28 +320,181 @@ void disable_all_steppers() {
   #endif
   #ifdef ACTION_ON_RESUMED
-    void host_action_resumed() { host_action(PSTR(ACTION_ON_RESUMED)); }
+    void host_action_resumed(const bool eol=true) { host_action(PSTR(ACTION_ON_RESUMED), eol); }
   #endif
   #ifdef ACTION_ON_CANCEL
-    void host_action_cancel() { host_action(PSTR(ACTION_ON_CANCEL)); }
+    void host_action_cancel(const bool eol=true) { host_action(PSTR(ACTION_ON_CANCEL), eol); }
   #endif
-  #ifdef ACTION_ON_FILAMENT_RUNOUT
-    void host_action_filament_runout(const bool eol/*=true*/) { host_action(PSTR(ACTION_ON_FILAMENT_RUNOUT), eol); }
+
+  #if ENABLED(HOST_PROMPT_SUPPORT)
+    void host_action_prompt(const char * const ptype, const bool eol=true) {
+      host_action(PSTR("prompt_"), false);
+      serialprintPGM(ptype);
+      if (eol) SERIAL_EOL();
+    }
+    void host_action_prompt_plus(const char * const ptype, const char * const pstr, const bool eol=true) {
+      host_action_prompt(ptype, false);
+      SERIAL_CHAR(' ');
+      serialprintPGM(pstr);
+      if (eol) SERIAL_EOL();
+    }
+    void host_action_prompt_begin(const char * const pstr, const bool eol=true) {
+      host_action_prompt_plus(PSTR("begin"), pstr, eol);
+    }
+    void host_action_prompt_button(const char * const pstr) {
+      host_action_prompt_plus(PSTR("button"), pstr);
+    }
+    void host_action_prompt_end() { host_action_prompt(PSTR("end")); }
+    void host_action_prompt_show() { host_action_prompt(PSTR("show")); }
   #endif
 
-#endif // HAS_ACTION_COMMANDS
+#endif  // HOST_ACTION_COMMANDS
+
+#if ENABLED(FILAMENT_RUNOUT_SENSOR)
+
+  void event_filament_runout() {
+    #if ENABLED(EXTENSIBLE_UI)
+      ExtUI::onFilamentRunout(ExtUI::getActiveTool());
+    #endif
+
+    static char tool;
+    #if NUM_RUNOUT_SENSORS > 1
+      tool = (char)int(active_extruder));
+    #else
+      tool = '0';
+    #endif
+    //action:out_of_filament
+    #if ENABLED(HOST_PROMPT_SUPPORT)
+      host_prompt_reason = PROMPT_FILAMENT_RUNOUT_TRIPPED;
+      host_action_prompt_end();
+      host_action_prompt_begin(PSTR("FilamentRunout T"), false);
+      SERIAL_CHAR(tool);
+      SERIAL_EOL();
+      host_action_prompt_show();
+    #endif
+
+    #if ENABLED(HOST_ACTION_COMMANDS)
+      if ( !runout.host_handling && strstr(FILAMENT_RUNOUT_SCRIPT, "M600") || strstr(FILAMENT_RUNOUT_SCRIPT, "M125") || (strstr(FILAMENT_RUNOUT_SCRIPT, "M25") && 
+      #if ENABLED(ADVANCED_PAUSE_FEATURE)
+        true
+      #else
+        false
+      #endif
+       ))
+        host_action_paused(false);
+      else {
+        // Legacy Repetier command to be used until newer version support standard dialog
+        // Will be removed at a later date when pause command also triggers dialog
+        SERIAL_ECHOPGM("//action:out_of_filament T");
+        SERIAL_CHAR(tool);
+        SERIAL_EOL();
+
+        host_action_pause(false);
+      }
+      SERIAL_ECHOPGM(" ACTION_REASON_ON_FILAMENT_RUNOUT");
```

Comment:
Why is this printing the literal string `" ACTION_REASON_ON_FILAMENT_RUNOUT"` to the host?